code/game/objects/structures/mirror.dm

```diff
@@ -9,74 +9,110 @@
 	var/shattered = 0
 
 
-/obj/structure/mirror/attack_hand(mob/user as mob)
+/obj/structure/mirror/attack_hand(mob/user)
+	choose(user, user)
+
+/obj/structure/mirror/proc/can_use(mob/living/user, mob/living/carbon/human/target)
 	if(shattered)
+		return FALSE
+	if(!ishigherbeing(user) || !ishuman(target))
+		return FALSE
+	if(!isturf(user.loc) || !isturf(target.loc))
+		return FALSE
+	if(!Adjacent(user) || !Adjacent(target))
+		return FALSE
+	if(user.incapacitated())
+		return FALSE
+	return TRUE
+
+/obj/structure/mirror/proc/choose(mob/living/user, mob/living/carbon/human/target)
+	if(!can_use(user, target))
 		return
-
-	if(ishuman(user))
-		var/mob/living/carbon/human/H = user
-		var/datum/role/vampire/V = isvampire(H)
-		if(isvampire(H))
-			if(!(VAMP_MATURE in V.powers))
+	var/mob/living/carbon/human/H = user
```

Comment:
this seems to make it so that you can change a vampire's hair in the mirror, but they can't work on someone else's hair (user instead of target)

Reply:
I'm not sure how to do this best while not making the current antag check even worse.
Right now without my changes you could force someone to change their hair with a mirror to check if they're a vampire, but with my changes you could check that by yourself by just dragging them onto a mirror, if I were to change this `user` to `target`